### PR TITLE
Cleanup - Initial Project Refactor

### PR DESCRIPTION
**Pull request by ssulei7**
* Updated `go.mod` to reflect source repository name
* Moved commit remap logic from `cmd/root.go` into `internal/commitremap`. Chances are some functions around file processing can be broken out of here as well, but we'll see.
* Removed `func main`, and did an inline function for the root cmd. I didn't see much of a point of adding a concrete function for the root command at this point in time 🤷🏽 
* Added initial unit test around parsing commit map
* Added unit test execution as part of CI workflow

### CODE REVIEW

**Comment by ssulei7** [comment]
@amenocal @kuhlman-labs please do an integration test and review the above. I have a bit more changes incoming regarding unit tests, but getting this initial PR out there. :)

File: internal/commitremap/commitremap.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	}
+
+	// Overwrite the original file with the updated data
+	err = os.WriteFile(filePath, updatedData, 0644)
+	if err != nil {
+		log.Fatalf("Error writing updated data: %v", err)
+	}
+}
+
+func replaceSHA(data interface{}, oldSHA string, newSHA string) {
```
Comment (ssulei7) [inline internal/commitremap/commitremap.go:98]: In a later PR, I would like to rework this function as I'm getting super spooked around the usage of empty interfaces here (e.g., any type)

Comment (amenocal) [reply internal/commitremap/commitremap.go:98]: @ssulei7 good catch!

File: internal/commitremap/commitremap.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+		// Process each file
+		for _, file := range files {
+			log.Println("Processing file:", file)
+
+			updateMetadataFile(file, commitMap)
+		}
+	}
+}
+
+func updateMetadataFile(filePath string, commitMap *[]CommitMapEntry) {
```
Comment (amenocal) [inline internal/commitremap/commitremap.go:67]: @ssulei7 should we move anything processing local files to it's own internal package ?

**Review by amenocal — APPROVED** [review]

File: internal/commitremap/commitremap.go (hunk 2 above)
Comment (ssulei7) [reply internal/commitremap/commitremap.go:67]: Probably would be a good idea.

Comment (kuhlman-labs) [reply internal/commitremap/commitremap.go:67]: @amenocal What would be the advantage of moving to it to an internal package?

File: internal/commitremap/commitremap.go (hunk 1 above)
Comment (kuhlman-labs) [reply internal/commitremap/commitremap.go:98]: Agreed

**Comment by kuhlman-labs** [comment]
@ssulei7 LGTM. My integration tests passed, I just added one commit to add in the issue_events file to be processed as well.

**Review by kuhlman-labs — APPROVED** [review]

File: internal/commitremap/commitremap.go (hunk 2 above)
Comment (amenocal) [reply internal/commitremap/commitremap.go:67]: @kuhlman-labs easier to find things and more organization

Comment (ssulei7) [reply internal/commitremap/commitremap.go:67]: @kuhlman-labs and adding on to @amenocal 's point, if for whatever reason if we need to add an additional feature or address a bug, it'll be a lot more structured to handle.

**Pull request merged**